### PR TITLE
DOCSP-25547-clarify-aws-auth

### PR DESCRIPTION
**Pull request by ccho-mongodb**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-25547>
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-25547-clarify-aws-admonition/fundamentals/authentication/mechanisms/#mongodb-aws

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/code-snippets/authentication/aws-env-variable.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 const { MongoClient } = require("mongodb");
 
-// Remember to add your AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY 
-// credentials to your environment variables.
-const clusterUrl = "<MongoDB cluster url>";
+// Remember specify your AWS credentials in environment variables.
```
Comment (rustagir) [inline source/code-snippets/authentication/aws-env-variable.js:3]:
```suggestion
// Remember to specify your AWS credentials in environment variables.
```

File: source/fundamentals/authentication/mechanisms.txt [hunk 2]
```diff
@@ -152,26 +152,33 @@ The driver checks for your credentials in the following sources in order:
 4. AWS EC2 endpoint. For more information, see `IAM Roles for Tasks
    <https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html>`_.
 
+.. important::
+
+   The driver only reads the credentials from the first method that it detects
+   in the order of the preceding list. For example, if you specify your
```
Comment (rustagir) [inline source/fundamentals/authentication/mechanisms.txt:158]:
```suggestion
   in the order given by the preceding list. For example, if you specify your
```

File: source/fundamentals/authentication/mechanisms.txt [hunk 3]
```diff
@@ -152,26 +152,33 @@ The driver checks for your credentials in the following sources in order:
...
+   AWS credentials in the connection string, it ignores any credentials
```
Comment (rustagir) [inline source/fundamentals/authentication/mechanisms.txt:159]: Suggestion: replace pronoun to disambiguate

```suggestion
   AWS credentials in the connection string, the driver ignores any credentials
```

**Review by rustagir — APPROVED** [review]
LGTM -- left some non-blocking comments for clarity

File: source/fundamentals/authentication/mechanisms.txt (hunk 2 above)
Comment (ccho-mongodb) [reply source/fundamentals/authentication/mechanisms.txt:158]: This suggestion seems more appropriate than the original, but I think the phrase is usually "as given by" so I'll use that.

**Pull request merged**